### PR TITLE
[Merged by Bors] - chore: split Order.Filter.Basic, creating Order.Filter.Finite

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
!bench

File: Mathlib/Tactic/Peel.lean [hunk 1]
```diff
@@ -3,8 +3,8 @@ Copyright (c) 2023 Jireh Loreaux. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jireh Loreaux
 -/
-import Mathlib.Order.Filter.Basic
 import Mathlib.Tactic.Basic
+import Mathlib.Order.Filter.Basic
```
Comment (Ruben-VandeVelde) [inline Mathlib/Tactic/Peel.lean:7]: I'm not going to insist that you sort the import blocks, but let's at least not make it worse :)
```suggestion
import Mathlib.Order.Filter.Basic
import Mathlib.Tactic.Basic
```

**Comment by Ruben-VandeVelde** [comment]
Impact on the longest pole: the path still goes from filters to finiteness, but no longer through `Order.Filter.Basic`. This seems alright to me.
```diff
-Analyzing Mathlib at 1f86e43e4838ee944676f80cab98f4c66c3ed0a9
+Analyzing Mathlib at ed5c0b2c6a7979201f63e349791e5efe7579b784
 | file                                                                 | instructions | cumulative | parallelism |
 | :------------------------------------------------------------------- | -----------: | ---------: | :---------: |
-| Mathlib                                                              |        11599 |    7440322 |    21.52    |
+| Mathlib                                                              |        11600 |    7357540 |    21.77    |
-| Mathlib.Analysis.CStarAlgebra.Module.Constructions                   |        82147 |    7428722 |    7.86     |
+| Mathlib.Analysis.CStarAlgebra.Module.Constructions                   |        82148 |    7345940 |    7.95     |
-| Mathlib.Analysis.CStarAlgebra.Module.Defs                            |        65758 |    7346575 |    7.93     |
+| Mathlib.Analysis.CStarAlgebra.Module.Defs                            |        65759 |    7263791 |    8.02     |
-| Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Order     |       244750 |    7280816 |    7.99     |
+| Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Order     |       244746 |    7198031 |    8.09     |
-| Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Instances |       361292 |    7036065 |    8.21     |
+| Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Instances |       361297 |    6953284 |    8.30     |
-| Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Unique    |       215790 |    6674773 |    8.47     |
+| Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Unique    |       215792 |    6591987 |    8.57     |
-| Mathlib.Analysis.Normed.Algebra.Spectrum                             |       150142 |    6458983 |    8.57     |
+| Mathlib.Analysis.Normed.Algebra.Spectrum                             |       150132 |    6376194 |    8.68     |
-| Mathlib.Analysis.Complex.Polynomial.Basic                            |        32407 |    6308841 |    8.66     |
+| Mathlib.Analysis.Complex.Polynomial.Basic                            |        32408 |    6226061 |    8.78     |
-| Mathlib.Analysis.Complex.Liouville                                   |        18738 |    6276434 |    8.22     |
+| Mathlib.Analysis.Complex.Liouville                                   |        18738 |    6193653 |    8.33     |
-| Mathlib.Analysis.Complex.CauchyIntegral                              |        52842 |    6257695 |    8.25     |
+| Mathlib.Analysis.Complex.CauchyIntegral                              |        52843 |    6174914 |    8.36     |
-| Mathlib.MeasureTheory.Integral.CircleIntegral                        |        75677 |    6204852 |    8.15     |
+| Mathlib.MeasureTheory.Integral.CircleIntegral                        |        75679 |    6122070 |    8.26     |
-| Mathlib.Analysis.SpecialFunctions.NonIntegrable                      |        16064 |    6129174 |    8.24     |
+| Mathlib.Analysis.SpecialFunctions.NonIntegrable                      |        16064 |    6046391 |    8.35     |
-| Mathlib.Analysis.SpecialFunctions.Log.Deriv                          |        50809 |    6113109 |    6.85     |
+| Mathlib.Analysis.SpecialFunctions.Log.Deriv                          |        50811 |    6030326 |    6.95     |
-| Mathlib.Analysis.SpecialFunctions.ExpDeriv                           |        47851 |    6062299 |    6.88     |
+| Mathlib.Analysis.SpecialFunctions.ExpDeriv                           |        47855 |    5979515 |    6.98     |
-| Mathlib.Analysis.Complex.RealDeriv                                   |        42524 |    6014447 |    6.77     |
+| Mathlib.Analysis.Complex.RealDeriv                                   |        42524 |    5931659 |    6.86     |
-| Mathlib.Analysis.Calculus.ContDiff.Basic                             |       406328 |    5971923 |    5.50     |
+| Mathlib.Analysis.Calculus.ContDiff.Basic                             |       406323 |    5889135 |    5.58     |
-| Mathlib.Analysis.Calculus.FDeriv.Mul                                 |       387781 |    5565594 |    5.78     |
+| Mathlib.Analysis.Calculus.FDeriv.Mul                                 |       387776 |    5482811 |    5.87     |
-| Mathlib.Analysis.Calculus.FDeriv.Analytic                            |       264226 |    5177812 |    6.13     |
+| Mathlib.Analysis.Calculus.FDeriv.Analytic                            |       264224 |    5095035 |    6.23     |
-| Mathlib.Analysis.Analytic.CPolynomial                                |       140595 |    4913586 |    6.13     |
+| Mathlib.Analysis.Analytic.CPolynomial                                |       140598 |    4830811 |    6.24     |
-| Mathlib.Analysis.Analytic.Constructions                              |       130377 |    4772991 |    6.26     |
+| Mathlib.Analysis.Analytic.Constructions                              |       130382 |    4690212 |    6.37     |
-| Mathlib.Analysis.Analytic.Composition                                |       164925 |    4642613 |    6.36     |
+| Mathlib.Analysis.Analytic.Composition                                |       164931 |    4559830 |    6.48     |
-| Mathlib.Analysis.Analytic.Basic                                      |       245605 |    4477688 |    6.55     |
+| Mathlib.Analysis.Analytic.Basic                                      |       245602 |    4394898 |    6.67     |
-| Mathlib.Analysis.Calculus.FormalMultilinearSeries                    |        82651 |    4232082 |    6.45     |
+| Mathlib.Analysis.Calculus.FormalMultilinearSeries                    |        82653 |    4149296 |    6.58     |
-| Mathlib.Analysis.NormedSpace.Multilinear.Curry                       |       212492 |    4149431 |    6.56     |
+| Mathlib.Analysis.NormedSpace.Multilinear.Curry                       |       212479 |    4066642 |    6.69     |
-| Mathlib.Analysis.NormedSpace.Multilinear.Basic                       |       294123 |    3936938 |    6.86     |
+| Mathlib.Analysis.NormedSpace.Multilinear.Basic                       |       294122 |    3854163 |    7.01     |
-| Mathlib.Analysis.NormedSpace.OperatorNorm.NormedSpace                |        62704 |    3642814 |    7.23     |
+| Mathlib.Analysis.NormedSpace.OperatorNorm.NormedSpace                |        62705 |    3560040 |    7.40     |
-| Mathlib.Analysis.NormedSpace.OperatorNorm.Bilinear                   |       172651 |    3580110 |    7.33     |
+| Mathlib.Analysis.NormedSpace.OperatorNorm.Bilinear                   |       172660 |    3497335 |    7.50     |
-| Mathlib.Analysis.NormedSpace.OperatorNorm.Basic                      |        89278 |    3407458 |    7.65     |
+| Mathlib.Analysis.NormedSpace.OperatorNorm.Basic                      |        89276 |    3324675 |    7.84     |
-| Mathlib.Topology.Algebra.Module.StrongTopology                       |       115236 |    3318179 |    7.51     |
+| Mathlib.Topology.Algebra.Module.StrongTopology                       |       115237 |    3235399 |    7.71     |
-| Mathlib.Topology.Algebra.Module.UniformConvergence                   |        13478 |    3202943 |    7.73     |
+| Mathlib.Topology.Algebra.Module.UniformConvergence                   |        13478 |    3120161 |    7.94     |
-| Mathlib.Analysis.LocallyConvex.Bounded                               |        49315 |    3189464 |    7.75     |
+| Mathlib.Analysis.LocallyConvex.Bounded                               |        49316 |    3106682 |    7.95     |
-| Mathlib.Analysis.Seminorm                                            |       159310 |    3140149 |    7.54     |
+| Mathlib.Analysis.Seminorm                                            |       159316 |    3057366 |    7.75     |
-| Mathlib.Analysis.LocallyConvex.Basic                                 |        36436 |    2980838 |    7.86     |
+| Mathlib.Analysis.LocallyConvex.Basic                                 |        36435 |    2898050 |    8.08     |
-| Mathlib.Analysis.Normed.Module.Basic                                 |        55192 |    2944402 |    7.60     |
+| Mathlib.Analysis.Normed.Module.Basic                                 |        55194 |    2861614 |    7.82     |
-| Mathlib.Analysis.Normed.Field.Lemmas                                 |        33275 |    2889209 |    7.70     |
+| Mathlib.Analysis.Normed.Field.Lemmas                                 |        33275 |    2806419 |    7.93     |
-| Mathlib.Analysis.Normed.Group.Uniform                                |        37153 |    2855934 |    6.99     |
+| Mathlib.Analysis.Normed.Group.Uniform                                |        37156 |    2773144 |    7.20     |
-| Mathlib.Topology.MetricSpace.Algebra                                 |        15537 |    2818780 |    6.89     |
+| Mathlib.Topology.MetricSpace.Algebra                                 |        15538 |    2735988 |    7.11     |
-| Mathlib.Topology.Algebra.SeparationQuotient.Basic                    |        37318 |    2803243 |    6.06     |
+| Mathlib.Topology.Algebra.SeparationQuotient.Basic                    |        37320 |    2720450 |    6.24     |
-| Mathlib.Topology.Algebra.Module.Basic                                |       288750 |    2765924 |    6.12     |
+| Mathlib.Topology.Algebra.Module.Basic                                |       288762 |    2683129 |    6.32     |
-| Mathlib.Topology.Algebra.UniformGroup.Defs                           |        51274 |    2477173 |    5.43     |
+| Mathlib.Topology.Algebra.UniformGroup.Defs                           |        51276 |    2394367 |    5.62     |
-| Mathlib.Topology.Algebra.Group.Basic                                 |       114992 |    2425898 |    5.49     |
+| Mathlib.Topology.Algebra.Group.Basic                                 |       115001 |    2343091 |    5.69     |
-| Mathlib.Topology.Algebra.Monoid                                      |        54732 |    2310906 |    4.62     |
+| Mathlib.Topology.Algebra.Monoid                                      |        54739 |    2228089 |    4.80     |
-| Mathlib.Topology.Algebra.MulAction                                   |        16131 |    2256174 |    4.44     |
+| Mathlib.Topology.Algebra.MulAction                                   |        16132 |    2173350 |    4.61     |
-| Mathlib.Topology.Algebra.ConstMulAction                              |        41982 |    2240043 |    4.45     |
+| Mathlib.Topology.Algebra.ConstMulAction                              |        41984 |    2157218 |    4.62     |
-| Mathlib.Topology.Algebra.Constructions                               |         8664 |    2198060 |    4.21     |
+| Mathlib.Topology.Algebra.Constructions                               |         8664 |    2115233 |    4.38     |
-| Mathlib.Topology.Homeomorph                                          |        50446 |    2189395 |    4.22     |
+| Mathlib.Topology.Homeomorph                                          |        50448 |    2106568 |    4.39     |
-| Mathlib.Topology.Support                                             |        21142 |    2138949 |    4.27     |
+| Mathlib.Topology.Support                                             |        21144 |    2056120 |    4.45     |
-| Mathlib.Topology.Separation.Basic                                    |        96475 |    2117806 |    4.01     |
+| Mathlib.Topology.Separation.Basic                                    |        96483 |    2034976 |    4.18     |
-| Mathlib.Topology.Compactness.Lindelof                                |        27174 |    2021331 |    4.06     |
+| Mathlib.Topology.Compactness.Lindelof                                |        27176 |    1938492 |    4.24     |
-| Mathlib.Topology.Compactness.SigmaCompact                            |        16736 |    1994157 |    4.10     |
+| Mathlib.Topology.Compactness.SigmaCompact                            |        16738 |    1911315 |    4.28     |
-| Mathlib.Topology.Compactness.LocallyCompact                          |         9497 |    1977421 |    4.12     |
+| Mathlib.Topology.Compactness.LocallyCompact                          |         9498 |    1894577 |    4.31     |
-| Mathlib.Topology.Compactness.Compact                                 |        45217 |    1967923 |    4.14     |
+| Mathlib.Topology.Compactness.Compact                                 |        45221 |    1885078 |    4.32     |
-| Mathlib.Topology.Bases                                               |        47836 |    1922705 |    3.99     |
+| Mathlib.Topology.Bases                                               |        47838 |    1839856 |    4.17     |
-| Mathlib.Topology.ContinuousOn                                        |        61122 |    1874868 |    4.06     |
+| Mathlib.Topology.ContinuousOn                                        |        61125 |    1792018 |    4.26     |
-| Mathlib.Topology.Constructions                                       |        80751 |    1813746 |    4.05     |
+| Mathlib.Topology.Constructions                                       |        80759 |    1730893 |    4.24     |
-| Mathlib.Topology.Maps.Basic                                          |        32743 |    1732994 |    4.11     |
+| Mathlib.Topology.Maps.Basic                                          |        32747 |    1650134 |    4.32     |
-| Mathlib.Topology.Order                                               |        38105 |    1700250 |    4.17     |
+| Mathlib.Topology.Order                                               |        38112 |    1617386 |    4.39     |
-| Mathlib.Topology.Defs.Induced                                        |         5241 |    1662145 |    4.25     |
+| Mathlib.Topology.Defs.Induced                                        |         5241 |    1579274 |    4.47     |
-| Mathlib.Topology.Basic                                               |        66856 |    1656904 |    4.26     |
+| Mathlib.Topology.Basic                                               |        66865 |    1574032 |    4.48     |
-| Mathlib.Order.Filter.AtTopBot                                        |        67922 |    1590047 |    4.25     |
+| Mathlib.Order.Filter.AtTopBot                                        |        67924 |    1507167 |    4.49     |
-| Mathlib.Order.Filter.Bases                                           |        40251 |    1522124 |    4.28     |
+| Mathlib.Order.Filter.Bases                                           |        40254 |    1439243 |    4.53     |
-| Mathlib.Order.Filter.Basic                                           |       100838 |    1481873 |    4.18     |
-| Mathlib.Data.Set.Finite.Lattice                                      |        18258 |    1381035 |    4.40     |
-| Mathlib.Data.Finite.Sigma                                            |         2825 |    1362777 |    4.35     |
-| Mathlib.Data.Fintype.Sigma                                           |        17411 |    1359952 |    4.29     |
-| Mathlib.Data.Finset.Sigma                                            |        12161 |    1342540 |    4.24     |
+| Mathlib.Data.Set.Countable                                           |        11280 |    1398988 |    4.41     |
+| Mathlib.Logic.Equiv.List                                             |        15986 |    1387708 |    4.41     |
+| Mathlib.Data.Finset.Sort                                             |        14914 |    1371721 |    4.32     |
+| Mathlib.Data.Finset.Max                                              |        26350 |    1356807 |    4.18     |
-| Mathlib.Data.Finset.Lattice.Fold                                     |        72839 |    1330378 |    4.24     |
+| Mathlib.Data.Finset.Lattice.Fold                                     |        72846 |    1330456 |    4.24     |
-| Mathlib.Data.Finset.Prod                                             |        18587 |    1257538 |    3.84     |
+| Mathlib.Data.Finset.Prod                                             |        18588 |    1257610 |    3.84     |
-| Mathlib.Data.Finset.Card                                             |        36204 |    1238951 |    3.89     |
+| Mathlib.Data.Finset.Card                                             |        36205 |    1239021 |    3.89     |
-| Mathlib.Data.Finset.Image                                            |        32979 |    1202746 |    3.97     |
+| Mathlib.Data.Finset.Image                                            |        32980 |    1202816 |    3.97     |
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**